librz/analysis/fcn.c

```diff
@@ -2338,8 +2352,19 @@ static void update_analysis(RzAnalysis *analysis, RzList /*<RzAnalysisFunction *
 			fcn->ninstr -= bb->ninstr;
 			rz_analysis_function_remove_block(fcn, bb);
 		}
+		RzList *bbs_temp = rz_list_new();
+		if (!bbs_temp) {
+			// If memory allocation failed, print an error message and return from the function
+			RZ_LOG_ERROR("Failed to allocate memory for bbs.\n");
+			return;
+		}
+
+		void **it;
+		rz_pvector_foreach (fcn->bbs, it) {
+			rz_list_append(bbs_temp, *it);
+		}
 
-		RzList *bbs = rz_list_clone(fcn->bbs);
+		RzList *bbs = rz_list_clone(bbs_temp);
```

Comment:
you don't need to clone. just name bbs_temp as bbs.

Reply:
On the other hand, you should convert rz_analysis_block_automerge to use RzPVector

Reply:
Ok